src/main.js

```diff
@@ -44,9 +45,71 @@ function ymlContentToConfig(ymlContent) {
   }
 
   if (configJson.resource) {
-    config.source = configJson.resource;
+    config.source = removeTrailingSlashes(configJson.resource);
+    config.multiple = false;
+    config.cacheComparator = compareArrays;
+    config.diffChecker = onlyInLeft;
+    config.resourceUpdater = (del, add) => updateResource(del, add, config.source);
+  } else if (configJson.resources) {
+    config.multiple = true;
+    config.resourceHostmap = configJson.resources.map((object) => {
+      return {
+        resource: removeTrailingSlashes(object.resource),
+        host: removeTrailingSlashes(object.host)
+      };
+    });
+    config.cacheComparator = (first, second, comparator) => {
+      const firstKeys = Object.keys(first);
+      const secondKeys = Object.keys(second);
+      if (firstKeys.length !== secondKeys.length) {
+        return false;
+      }
+      const result = firstKeys.every((entry) => secondKeys.includes(entry));
+      if (!result) {
+        return false;
+      }
+      return firstKeys.every((entry) => compareArrays(first[entry], second[entry], comparator));
+    };
+    config.diffChecker = (left, right, cmp) => {
+      // the assumption is made that the structures of left and right are the same
+      const leftKeys = Object.keys(left);
+      const out = {};
+      leftKeys.forEach((key) => out[key] = onlyInLeft(left[key], right[key], cmp));
+      return out;
+    };
+    config.resourceUpdater = async (del, add) => {
```

Comment:
Move this function to a separate function. I know it's probably only used here. But it will split this function up in smaller ones and make it more readable. Additionally, it will force use to add documentation to the function as well 😉